### PR TITLE
Fix exporting input data field prior transaction parsing

### LINKED ISSUE #18: eth_getTransactionByHash returns incorrect input data.
I see that `eth_getTransactionByHash` is still WIP, but opening this issue as this functionality is needed in the bridge web app.

Here is an erc20 token transfer transaction made with MetaMask: 
```
curl https://testnet.aurora.dev -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params": ["0x0861c33b5319e5190fabc6426ba36583fa379a9b0a887a6110f7bac6f332f42e"],"id":1}'
```
Result:
```
{"jsonrpc":"2.0","id":1,"result":{"blockNumber":"0x3b41465","blockHash":"0xa8f673e7fa5310eb16e919823ba51f2a701e6768577f4942241bc13f5e59b1e0","transactionIndex":"0x0","hash":"0x0861c33b5319e5190fabc6426ba36583fa379a9b0a887a6110f7bac6f332f42e","from":"0xe6220257d157ec7b481290fd10d2037cf0e83ea5","to":"0xf93cd0e464f74c240d8ebb7ed55ce6b43452f913","gas":"0x6691b7","gasPrice":"0x0","nonce":"0x137","value":"0x0","input":"0x","v":"0x9c8a82c9","r":"0x524a2cb20c479a30a5f500fdf379a5cc835b7971310a4b646fc8ccd6c7112521","s":"0x71a9a58109b4f326f7e848f98d90db440abbf72e4993989d5277d775a222b49a"}}
```
The `input` field should not be empty.

I suspect this is also causing the following error with ethersjs:
```js
await provider.getBlockWithTransactions(62109188)
```
```
Error: invalid hash (argument="value", value=undefined, code=INVALID_ARGUMENT, version=providers/5.4.1)
    at Logger.makeError (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-logger-npm-5.4.0-a27b22da34-4b01d88f95.zip/node_modules/@ethersproject/logger/lib/index.js:187:21)
    at Logger.throwError (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-logger-npm-5.4.0-a27b22da34-4b01d88f95.zip/node_modules/@ethersproject/logger/lib/index.js:196:20)
    at Logger.throwArgumentError (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-logger-npm-5.4.0-a27b22da34-4b01d88f95.zip/node_modules/@ethersproject/logger/lib/index.js:199:21)
    at Formatter.hex (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:180:23)
    at Formatter.data (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:183:27)
    at Function.Formatter.check (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:373:40)
    at Formatter.transactionResponse (/Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:283:32)
    at /Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:412:29
    at Array.forEach (<anonymous>)
    at /Users/pa/js_workspace/github/aurora-is-near/rainbow-bridge-client/.yarn/cache/@ethersproject-providers-npm-5.4.1-cc2f084447-d9bd8288db.zip/node_modules/@ethersproject/providers/lib/formatter.js:411:19 {
  reason: 'invalid hash',
  code: 'INVALID_ARGUMENT',
  argument: 'value',
  value: undefined,
  checkKey: 'transactions',
  checkValue: [
    {
      blockNumber: '0x3b3b604',
      blockHash: '0x9e3a4517a6cf62ffcb473e3db44740e39825f2dba4859ca06ce53885dcd355fe',
      transactionIndex: '0x0',
      hash: '0x66111f31f83aee4cedc4bee8aeabc608631557e0fd7397767f1f930801bfd228',
      from: '0xe6220257d157ec7b481290fd10d2037cf0e83ea5',
      to: '0xf93cd0e464f74c240d8ebb7ed55ce6b43452f913',
      gas: '0x186a0',
      gasPrice: '0x0',
      nonce: '0x136',
      value: '0x0',
      input: null,
      v: '0x9c8a82ca',
      r: '0x11cd13fae29559601e4d8e10eb93e1ebfeaa0514334bc64a27dfb8483602f9d0',
      s: '0xe3862cbf7e0d1aab3fba798a65f23e87b48a1063fa8a712bfdb8455a8f03faf',
      confirmations: 24339,
      gasLimit: '0x186a0'
    }
  ]
}
```
The transaction hashes cannot be verified due to the input data not begin present.

### PR DESCRIPTION
**Pull request by 0x3bfc**
closes #18

### CODE REVIEW

File: src/transaction.ts [hunk 1]
```diff
@@ -143,7 +143,7 @@ export class Transaction {
       gas: intToHex(this.gasLimit),
       to: this.to.isSome() ? this.to.unwrap().toString() : null,
       value: intToHex(this.value),
-      input: bytesToHex(this.data),
```
Comment (artob) [inline src/transaction.ts:146]: Surely this shouldn't change at all?

File: src/transaction.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { AccountID, Address } from './account.js';
 import NEAR, { NEARTransaction } from './near.js';
 import { None, Option, Some, U64, U256 } from './prelude.js';
 import { SubmitResult } from './schema.js';
-import { base58ToBytes, base58ToHex, bytesToHex, intToHex } from './utils.js';
```
Comment (antonpaisov) [inline src/transaction.ts:7]: `bytesToHex` should also remain as per @artob comment below

**Comment by artob** [comment]
@0x3bfc Since we now know the `Buffer.from(hexstring, 'hex'),` construct to be buggy, please check also the rest of the code to see if there perhaps are other places where that should be changed to `hexToBytes()`.

**Comment by 0x3bfc** [comment]
> @0x3bfc Since we now know the `Buffer.from(hexstring, 'hex'),` construct to be buggy, please check also the rest of the code to see if there perhaps are other places where that should be changed to `hexToBytes()`.

All good!

**Review by antonpaisov — APPROVED** [review]

File: lib/transaction.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import { AccountID, Address } from './account.js';
 import { None, Some } from './prelude.js';
 import { SubmitResult } from './schema.js';
-import { base58ToBytes, base58ToHex, bytesToHex, intToHex } from './utils.js';
+import { base58ToBytes, base58ToHex, bytesToHex, hexToBytes, intToHex, } from './utils.js';
```
Comment (antonpaisov) [inline lib/transaction.js:5]: do we need the extra `,` after `intToHex` @0x3bfc ?

Comment (0x3bfc) [reply lib/transaction.js:5]: This was added by the prettifier

**Review by artob — APPROVED** [review]

**Pull request merged**